src/rf/apps/workers/process.py

```diff
@@ -0,0 +1,221 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from apps.core.models import LayerImage, LayerMeta
+from django.conf import settings
+
+import time
+import json
+import boto.sqs
+from boto.sqs.message import Message
+
+TIMEOUT_SECONDS = (60 * 10)  # 10 Minutes.
+STATUS_VALIDATED = 'validated'
+STATUS_REPROJECTED = 'reprojected'
+STATUS_PROCESSING = 'processing'
+STATUS_FAILED = 'failed'
+STATUS_COMPLETE = 'complete'
+
+JOB_VALIDATE = [
+    'ObjectCreated:CompleteMultipartUpload',
+    'ObjectCreated:Put',
+    'ObjectCreated:Post'
+]
+JOB_REPROJECT = 'reproject'
+JOB_HANDOFF = 'emr_handoff'
+JOB_TIMEOUT = 'emr_timeout'
+
+MIN_WAIT = 1  # Second.
+MAX_WAIT = 60  # Seconds.
+
+
+class QueueProcessor(object):
+    """
+    Encapsulates behavior for connecting to an SQS queue and moving image data
+    throught the processing pipeline by adding new messages to the same queue.
+    """
+    def __init__(self):
+        self.q = self.get_queue()
+        self.wait = MIN_WAIT
+
+    def start_polling(self):
+        while True and self.q:
+            messages = self.q.get_messages(message_attributes=['All'])
```

Comment:
If we're only planning to deal with one message at a time, maybe we can use `read()` vs. `get_messages()`? Makes the later use of `write()` feel more intuitive.